### DISCUSSION TITLE
Compile and (manually) install PyRx for BCAD v25

### DISCUSSION THREAD

**Discussion by schoeller** [CEXT-Dan/PyRx#120] [General]
Thanks for PyRx.

In order to be able to test potential contributions (e.g. stubs), one must manually test before creating a pull request. Below is a walkthrough in order to document for others to use.

Requirements: MS W11, BCAD 25.0.3, x64 hardware

1. Install Python 3.12 (https://www.python.org/ftp/python/3.12.7/python-3.12.7-amd64.exe)

2. Install Visual Studio 2019 community edition including BuildingTools (v142)

3. Install Boost 1.8.5 for VC v142 (https://sourceforge.net/projects/boost/files/boost-binaries/1.85.0/boost_1_85_0-msvc-14.2-64.exe/download)

4. Install WxWidgets as [per download provided](https://drive.google.com/file/d/1ukMDR7qCfgEnD62-frIuph73dr-POv_9/view?usp=sharing) and extract into `C:\Program Files\wxWidgets-3.2.0`

6. Install WxPython on the PowerShell using `pip install wxpython`

7. Install the BCAD SDK BRX25.1.05.0 (https://www.bricsys.com/en-eu/developer-resources)

8. Set environment variables as follows

```
ARX2025=C:\Program Files\Bricsys\BRXSDK\BRX25.1.05.0
WXWIN=c:\Program Files\wxWidgets-3.2.0
BOOST=C:\Program Files\Boost_1_85_0
Py312=C:\Users\Sebastian Schoeller\AppData\Local\Programs\Python\Python312
WXPYWIN=C:\Users\Sebastian Schoeller\AppData\Local\Programs\Python\Python312Lib\site-packages\wx
BRX25=C:\Program Files\Bricsys\BRXSDK\BRX25.1.05.0
```

9. Fetch source code for PyRx
10. Modify `RxLoaderV25.0.vcxproj` and `PyRxV25.0.vcxproj` and change strings

```
inc-$(PlatformName)
lib-$(PlatformName)
```

to

```
$(BRX25)\inc$(PlatformArchitecture)
$(BRX25)\lib$(PlatformArchitecture)
```

11. Open `PyRx.sln` in Visual Studio 2019 Community Edition
12. Compile *debug* manually PyRxV25.0 and RxLoaderV25.0
<img src="https://github.com/user-attachments/assets/b65bf16b-135c-41af-aecf-7b7c0027e5b9" width=300>

13. Load module using _APPLOAD from Bin-folder

```
GitHub\PyRx\Bin\PyRxV25.0\x64\Debug\PyRxV25.0.brx
GitHub\PyRx\Bin\PyRxV25.0\x64\Debug\RxLoaderV25.0.brx
```

<img src="https://github.com/user-attachments/assets/5c702492-98f0-4a00-833d-620dcdbfd6f0" width=480>

One should receive a message similar to below

<img src="https://github.com/user-attachments/assets/48208c93-29c3-4e82-ad01-8b5ad76b5537" width=240>

The stub file should thus be generated in

```
GitHub\PyRx\PyRxStubs
```

14. Copy `PyBrxCv.pyi`, `PyRxV25.0.brx`, `RxLoaderV25.0.brx` into the installation directory wherever required.

Best

Sebastian

05.11.2024: Changed VS to 2019
11.11.2024: Changed WXWIN-instructions; changed load order
18.05.2025: When switching to Visual Studio 2022 (BuildingTools v142 to v143) the linker path must be adjusted for `BOOST` as follows:
`$(BOOST)\lib64-msvc-14.2` => `$(BOOST)\lib64-msvc-14.3` in `PyRxV25.0.vcxproj`

**Comment by CEXT-Dan** [comment]
Item 4, $(WXWIN) as noted in BUILDENV.txt. I have provided a wxWidgets build here.
https://drive.google.com/file/d/1ukMDR7qCfgEnD62-frIuph73dr-POv_9/view?usp=sharing

the issue could be that when linking, the .LIB and .DLL names must match the target. The download is “wxmsw32u_core_vc14x_x64.dll”
whereas in AppData\Local\Programs\Python\Python312\Lib\site-packages\wx, its “wxmsw32u_core_vc140_x64.dll”

you can't jut rename the files, however, you can compile wxWidgets yourself, but you have to go through every project and change the .LIB and .DLL names to exactly match what’s in wxPython.
Not hard, just tedious 

Another thing is, I see is you should be loading RxLoaderV25.0.brx in _APPLOAD and not PyRxV25.0.brx, unless you already have a path to wxPython 

There’s a cool app here. https://github.com/lucasg/Dependencies, You can target PyRxV25.0 and see your actual dependencies

![depends](https://github.com/user-attachments/assets/06e64e23-02d5-4b5f-8d60-278bed63953f)

**Reply by schoeller** [reply]
@CEXT-Dan : works like a charmer. Thanks a ton.
<img src="https://github.com/user-attachments/assets/0da8595c-9355-463b-ab54-f190456d16c7" width=480>

**Comment by CEXT-Dan** [comment]
Just to add. From a development perspective, it may be best to uninstall the user package, and work from the solution directory. just load from PyRxGit\Bin
Reason, stub generator and tests are designed to work in the solution dir. 
It’s kind of a pain switching back and forth though, I.e. you may have to change the stub path in vs code. 
but it's easier than copying stubs back an forth

**Reply by CEXT-Dan** [reply]
... and if you want to debug the C++ code, it’s probably best to load/run from the solution dir
plus debug builds are much faster to compile :)

**Comment by CEXT-Dan** [comment]
> 18.05.2025: When switching to Visual Studio 2022 (BuildingTools v142 to v143) the linker path must be adjusted for BOOST as follows:

Bricscad is still 142 build tools It should still $(BOOST)\lib64-msvc-14.2
When you open visual studio, it will sometimes ask if you want to update, always choose no
